backend/dal/Services/Admin/Concrete/BuildingService.cs

```diff
@@ -124,6 +124,19 @@ public override Building Add(Building entity)
         {
             entity.ThrowIfNull(nameof(entity));
 
+            if (entity.Agency != null)
```

Comment:
This is to ensure we don't attempt to add new related objects when they are attached.  I am not 100% sure this is required, but was running into an issue at one point that this resolved.